### ISSUE TITLE
Xamarin.Forms Picker System.ArgumentException in UWP

### ISSUE THREAD

**Issue by vsfeedback** [xamarin/Xamarin.Forms#8336]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/791396/xamarinforms-picker-systemargumentexception-in-uwp.html)._

---
This issue was introduced with Xamarin 4.3.0.908675
With the following Xaml code. (The ViewModel is shown futher down)

`&lt;Picker Grid.Row=&quot;2&quot; x:Name=&quot;MyPicker&quot; ItemsSource=&quot;{Binding PickerItems}&quot; SelectedItem=&quot;{Binding SelectedPickerItem}&quot; &gt;&lt;/Picker&gt;`

When assigning a ViewModel to the BindingContext of a Picker as follows....

```
InitializeComponent();
Device.BeginInvokeOnMainThread(() =&gt;
{          
      MyPicker.BindingContext = new PickerViewModel();
});
```
results in an System.ArgumentException; (see attached)

It is not specifically setting the BindingContext that cause the issue. It is anytime the Picker's values in the ViewModel are assigned within the BeginInvokeOnMainThread.
So if you remove the constructor in the PickerViewModel so that the PickerItems is not initialized, and create the ItemsSource as follows in the Page Constructor, the same exception will be thrown.

```
InitializeComponent();
var pvm = new PickerViewModel();
MyPicker.BindingContext = pvm; // Assign the Binding Context here
Device.BeginInvokeOnMainThread(() =&gt;
{     
     // Initialize the Picker's ItemSource and SelectedItem bound properties here in the BeginInvoke.     
      pvm. PickerItems = new List&lt;string&gt;() { &quot;Item One&quot;, &quot;Item Two&quot;, &quot;Item Three&quot; };
      pvm. SelectedPickerItem = pvm. PickerItems[1];
});
```

If the BindingContext and the other property initialization is outside if the BeginInvokeOnMainThread then it does not throw an exception (it worked either way in Xamarin.Forms 4.2

The PickerViewModel is defined as 

```
public class PickerViewModel : INotifyPropertyChanged
    {

public PickerViewModel()
        {
            PickerItems = new List&lt;string&gt;() {&quot;Item One&quot;, &quot;Item Two&quot;, &quot;Item Three&quot;};
            SelectedPickerItem = PickerItems[1];
        }

private string _selectedPickerItem;

public string SelectedPickerItem
        {
            get =&gt; _selectedPickerItem;
            set
            {
                _selectedPickerItem = value;
                OnPropertyChanged();
            }
        }

private List&lt;string&gt; _pickerItems;

public List&lt;string&gt; PickerItems
        {
            get =&gt; _pickerItems;
            set
            {
                _pickerItems = value;
                OnPropertyChanged();
            }
        }

public event PropertyChangedEventHandler PropertyChanged;

protected virtual void OnPropertyChanged([CallerMemberName] string propertyName = null)
        {
            PropertyChanged?. Invoke(this, new PropertyChangedEventArgs(propertyName));
        }
    }
```

---
#### Visual Studio Feedback System on 10/23/2019, 07:40 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by hartez** [comment]
This is a duplicate of #8167. Should be fixed by #8311.

**Comment by samhouts** [comment]
closed by #8311
